src/nix-collect-garbage/nix-collect-garbage.cc

```diff
@@ -76,19 +76,31 @@ static int _main(int argc, char * * argv)
             return true;
         });
 
+        if (removeOld && dryRun)
+        {
+            // removeOldGenerations does not record any information
+            // about which generations would be deleted if dryRun is true, so
+            // we cannot get an accurate list of store paths to be deleted.
+            std::cerr << "error: Sorry, the --dry-run option is not yet "
+                "compatible with deleting generations." << std::endl;
```

Comment:
This should be `throw UsageError("bla bla");`.

Reply:
Thanks for taking a look!  I amended my commit so it uses `UsageError` now.